### ISSUE TITLE
request help:  Support  Backend for Frontend 

### ISSUE THREAD

**Issue by aaron-codes** [apache/apisix#4761]
### Issue description
能否支持下图中的 API 编排能力
![image](https://user-images.githubusercontent.com/3776267/128335580-8fb3b2db-6dbf-4a88-93ae-8331db0c79a3.png)

**Comment by tokers** [comment]
It's not a good way to support it in Apache APISIX but you can implement it in a custom plugin.

**Comment by juzhiyuan** [comment]
see https://apisix.apache.org/docs/apisix/plugins/batch-requests/

Now we support setting the same `Query` for different requests, does it make sense to you?

**Comment by membphis** [comment]
please change the title to English @aaron-codes

**Comment by abbluiz** [comment]
I think that is a feature I would like to see too. Other API gateways call it Sequential Proxy or chained backend requests. It's not the same as batch requests, because sometimes I need the response from a backend to be included in the request of the next backend.

**Comment by spacewander** [comment]
@abbluiz 
Maybe the batch-request plugin can serve this purpose?

**Comment by tzssangglass** [comment]
the batch-request plugin can sent more multiple requests, but the "Backend for Frontend" here also need to handle responses of multiple requests.

**Comment by abbluiz** [comment]
@spacewander @tzssangglass 

As I mentioned before, the batch-requests would not fully serve this purpose, because yes it would need to handle a few responses, if some request on the pipeline depends on the response of other requests.

Maybe it will become clear for you if you check the implementation of chained requests other API gateways have done, such as KrakenD:
https://www.krakend.io/docs/endpoints/sequential-proxy/

**Comment by spacewander** [comment]
I see. To support it, currently we have to write a custom plugin.

**Comment by spacewander** [comment]
@18300618319 suggested that we can study the API orchestration from the other gateways in https://github.com/apache/apisix/issues/5076

**Comment by 18300618319** [comment]
fizz-gateway这个API编排挺好用的




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年9月17日(星期五) 上午9:44
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [apache/apisix] request help:  Support  Backend for Frontend  (#4761)






@18300618319建议我们可以研究来自其他网关的 API 编排#5076

—
你收到这个是因为你被提到了。
直接回复此邮件，在 GitHub 上查看或取消订阅.
GitHub Mobile 上的分流通知监督办或机器人.

**Comment by abbluiz** [comment]
@spacewander 
How would I write a custom plugin for this? Would you recommend using the `ext-plugin-post-req` with Python, for example? Could I make additional request inside this external plugin?

**Comment by spacewander** [comment]
Yes, you can make additional request inside the external plugin.
There is a real-world example (not in Python): https://github.com/swisscom/apisix-opa-plugin/blob/master/auth.go

Or you can do it in the Lua, like https://github.com/apache/apisix/blob/master/apisix/plugins/forward-auth.lua

**Comment by abbluiz** [comment]
@spacewander Interesting. I will be trying that out in the future.

**Comment by Baoyuantop** [comment]
If there is still a problem, please open it again.
